src/main/java/org/eclipse/emfcloud/jackson/databind/property/EObjectFeatureProperty.java

```diff
@@ -52,12 +57,28 @@ public EObjectFeatureProperty(final EStructuralFeature feature, final JavaType t
       this.defaultValues = OPTION_SERIALIZE_DEFAULT_VALUE.enabledIn(features);
    }
 
+   private static final EDataTypeDeserializer EDATA_TYPE_DESERIALIZER = new EDataTypeDeserializer();
+
+   private boolean shouldUseEFactory(final EDataType dataType) {
+      return dataType.isSerializable() && dataType.getEPackage() != EcorePackage.eINSTANCE
+         && (!EDataTypeDeserializer.isJavaLangType(dataType));
+   }
+
+   private JsonDeserializer<Object> findValueDeserializer(final DeserializationContext ctxt)
+      throws JsonMappingException {
+      if ((!feature.isMany()) && feature instanceof EAttribute
+         && shouldUseEFactory(((EAttribute) feature).getEAttributeType())) {
+         return EDATA_TYPE_DESERIALIZER;
```

Comment:
I'd like to avoid using static instances of concrete classes like this as it makes it impossible for adopters to avoid it's usage. At the moment deserializers are registered through the EMFDeserializers class so I think we should use the registration mechanism to retrieve the deserializer we want:
```
ctxt.findContextualValueDeserializer(ctxt.getTypeFactory().constructType(EcoreType.DataType.class), null)
```

Reply:
Same as above: Sounds like a good idea!